tracker/sm-website-update-visitor.js

```diff
@@ -1,147 +1,153 @@
+(function(sm) {
+
+  updateVisitorCustomAttributes = function(attributeName, attributeValue){
+    getCurrentVisitor(salemove).then(function(visitor) {
+      visitor.custom_attributes[attributeName] = attributeValue
+      $.ajax({
+        type: "POST",
+        beforeSend: function (request)
+          {
+            requestHeaders = sm.salemoveApi.getRequestHeaders();
+              Object.keys(requestHeaders).forEach(function(key) {
+              request.setRequestHeader(key, requestHeaders[key]);
+            });
+          },
+        url: "https://api.salemove.com/visitor",
+        data:
+          {
+            'note_update_method': 'append',
+            'custom_attributes': visitor.customAttributes
+          }
+        });
+      }, function(error) {
+        console.log('An error occurred: ', error);
+      }
+    );
+  }
+
+  updateVisitorInformation = function(attributeName, attributeValue){
+    getCurrentVisitor(salemove).then(function(visitor) {
+      visitor[attributeName] = attributeValue;
+      delete visitor.id;
+      delete visitor.href;
+      $.ajax({
+        type: "POST",
+        beforeSend: function (request)
+          {
+            requestHeaders = sm.salemoveApi.getRequestHeaders();
+              Object.keys(requestHeaders).forEach(function(key) {
+              request.setRequestHeader(key, requestHeaders[key]);
+            });
+          },
+        url: "https://api.salemove.com/visitor",
+        data: visitor
+        });
+      }, function(error) {
+        console.log('An error occurred: ', error);
+      }
+    );
+  }
+
+
+  getCurrentVisitor = function() {
+
+    return new Promise(function(resolve, reject) {
+     var xhr = new XMLHttpRequest();
+     var API_DOMAIN = "api.salemove.com"
```

Comment:
I tested it in my local environment, it was clearer what part i needed to change. In a later improvement they can be extracted into a configuration object and they can be injected.